### ISSUE TITLE
Error: Invariant Violation: Server-side <Router>s need location, branch, params,

### ISSUE THREAD

**Issue by swennemans** [bananaoomarang/isomorphic-redux#4]
I'm following your tutorial at https://medium.com/@bananaoomarang/handcrafting-an-isomorphic-redux-application-with-love-40ada4468af4 and it looks very cool, thanks for sharing. But I'm encountering a problem.

Right after the part that states: 

> It still won’t look very interesting, but visiting http://localhost:8080/ should work without errors. Routing’s all set, and we’re ready for some Redux action.

I still encounter an error. The error is:
`Error: Invariant Violation: Server-side <Router>s need location, branch, params, and components props. Try using Router.run to get all the props you need`

However, after implementing everything upto `shared/lib/fetchComponentData.js` it does work as expected. Any idea why?

**Comment by bananaoomarang** [comment]
I think I've have just updated the tutorial to fix this error. The new code has:

```
if(!routeState) return res.status(404).end('404');
```

Before it tries to render `<Router>`. My theory for what was happening is:

Browser requests `favicon.ico`, which isn't a handled route.
Server tries to render `<Router>` when `routeState` is null
Kaboom

Then later in the tutorial a check is added, and it doesn't try to render unless `routeState` is truthy.

tl;dr is that if you add the 404 check this shouldn't happen anymore.

I think React Router has a way of handling 404s with a custom component, but I couldn't get it working just now in the `1.0.0beta` the tutorial uses. Hope this fixes the issue!

**Comment by swennemans** [comment]
Cool thanks :) 

Thanks again for making this tutorial. All those isomorphic/universal starters are great but sometimes too magical because you didn't write the code. In a tutorial form it's very nice because it builds your understanding.
